src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2092,17 +2104,33 @@ public override void AfterTransactionBegin(ITransaction tx)
 			using (new SessionIdLoggingContext(SessionId))
 			{
 				CheckAndUpdateSessionStatus();
-				Interceptor.AfterTransactionBegin(tx);
+
+				if (!_transactionCoordinatorShared || Interceptor != ConnectionManager.Session.Interceptor)
```

Comment:
Interceptor is not here only for transaction. All the other cases are still called. But when the transaction is shared, does it make sens to notifies the same interceptor multiple time of the same transaction beginning? If the session was supplied as an argument, why not, but that is not the case.